### ISSUE TITLE
Missing columns in Organize Fields transformation after Outer Join (New in 7.4)

### ISSUE THREAD

**Issue by stwellman** [grafana/grafana#31104]
**What happened**:
I used Outer Join transformation to join 3 queries.  this worked correctly and appended a number to common column names to prevent duplicates (i.e. Time 1, Time 2, Time 3).  I then tried to use the Organize Fields transformation to hide the redundant columns, but it only list the original column name (i.e. Time).  You can't hide individual columns.  You can only hide all or none.  This is new behavior in 7.4.  It worked as expected in 7.3.  

May be related to this issue that was fixed in 7.2: https://github.com/grafana/grafana/issues/24744

Screenshots:
![table transform](https://user-images.githubusercontent.com/16233453/107533769-e2365a00-6b84-11eb-8777-3eba30410153.png)
![queries](https://user-images.githubusercontent.com/16233453/107533790-e82c3b00-6b84-11eb-99cc-4313ae8d4296.png)

This is the behavior in 7.3
![image001 (1)](https://user-images.githubusercontent.com/16233453/107533719-d21e7a80-6b84-11eb-8ec2-5b4702aaed74.png)

**What you expected to happen**:
I expected the column names in Organize Fields to match the post-join columns as shown on the table (i.e. Time 1, Time 2, etc.), and allow you to hide individual columns. 

**How to reproduce it (as minimally and precisely as possible)**:
Add 2 metrics with common columns/labels.  Outer join on a common column.  Try to hide only one of the duplicate columns.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.4
- Data source type & version:  Prometheus 2.23.0-rc.0
- OS Grafana is installed on:  Ubuntu 20.10
- User OS & Browser: Windows 10, Chrome 88.0.4324.150
- Grafana plugins: none
- Others:

**Comment by svet-b** [comment]
Bizarrely, this appears to still be an issue when using the [GraphQL datasource](https://grafana.com/grafana/plugins/fifemon-graphql-datasource/). Confirmed in several Grafana versions, including 7.5.4.

I've been staring at the Transform debugger for a while, trying to work out what's going on - and why the behavior for this datasource is different - but am drawing a complete blank.

Here's a minimal reproducer:
1. Spin up a Grafana 7.5.4 container with the GraphQL datasource plugin installed:
```
docker run -d -p 3000:3000 --name=grafana_gql -e "GF_INSTALL_PLUGINS=fifemon-graphql-datasource" grafana/grafana
```
2. Create a GraphQL datasource, which points to the (public) URL https://bahnql.herokuapp.com/graphql. Leave all settings (including name) set to defaults.
3. Import the dashboard https://grafana.com/grafana/dashboards/14341, which should yield something like this:
<img width="679" alt="Screen Shot 2021-04-30 at 2 42 12 PM" src="https://user-images.githubusercontent.com/12831860/116739929-45849b00-a9c2-11eb-81a9-6eeb370ed0b9.png">

The dashboard has a single table panel with two queries. Each of them pulls the live train arrivals for a different train station. Each of the queries yields two fields called `platform` and `trainNumber`. If we do an outer join on the `platform` field, the `trainNumber` fields do _not_ get given unique names (with a numbered suffix) but are instead still both called `trainNumber`, which leads to all the problems in the original issue report.

I've played around quite a bit, but the only criterion for reproduction I've established so far is that it happens with all data from this datasource, but not with data from others. Happy to look at it from the datasource perspective of course, but would be good to have some idea as to the potential root cause. Cc @retzkek .
